src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,67 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints(dist.entry_points)
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    @staticmethod
+    def _parse_entrypoint(entrypoint: str) -> Tuple[str, str]:
+        """
+        Extract the entry point and name from the string.
+
+        .. code-block:: python
+
+            WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
+            # returns ('hello-world', 'timmins:hello_world')
+
+            WritePyproject._parse_entrypoint("hello-world")
+            # throws ValueError: Entry point 'hello-world' is not of the form 'name = target'
+
+        :param: entrypoint  The entry point string, of the form
+                            "entrypoint = module:function" (whitespace optional)
+        :returns:           A two-element `tuple`, first element is the entry point name, second element is the target
+                            (module and function name) as a string.
+        :raises ValueError: An equals (`=`) character was not present in the entry point string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
```

Comment:
This comment is no longer needed now that the expected input format is described in the documentation (and the examples).

```suggestion
```